### PR TITLE
manifest: add registry name to docker images

### PR DESCRIPTION
**Pull request by zalsader**
Adds a registry name to rook images

**Issue resolved by this Pull Request:**
Resolves: #14545

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release. <-- is this change notable enough to be added here?

### CODE REVIEW

File: deploy/examples/images.txt [hunk 1]
```diff
@@ -8,4 +8,4 @@
  registry.k8s.io/sig-storage/csi-provisioner:v4.0.1
  registry.k8s.io/sig-storage/csi-resizer:v1.10.1
  registry.k8s.io/sig-storage/csi-snapshotter:v7.0.2
- rook/ceph:master
+ docker.io/rook/ceph:master
```
Comment (travisn) [inline deploy/examples/images.txt:11]: With this change, the sort order of the images changed, looks like the CI expects it at the top of the file now

**Review by travisn — CHANGES_REQUESTED** [review]

**Review by travisn — CHANGES_REQUESTED** [review]
Could you add this change to the PendingReleaseNotes.md? While very small, I want to make sure it's in the release notes for 1.15.

The helm upgrade CI is failing. There is a helm upgrade test where the `docker.io` repo needs to be added to the check:
```
2024-08-09 17:41:25.498119 D | exec: Running command: /tmp/rook-tests-scripts-helm/helm upgrade rook-ceph-cluster /home/runner/work/rook/rook/deploy/charts/rook-ceph-cluster --namespace upgrade -f values-test.yaml
2024-08-09 17:41:26.741896 I | testutil: helm chart "rook-ceph-cluster" installed successfully after 1 attempt(s)
    ceph_upgrade_test.go:383: 
        	Error Trace:	/home/runner/work/rook/rook/tests/integration/ceph_upgrade_test.go:383
        	            				/home/runner/work/rook/rook/tests/integration/ceph_upgrade_test.go:137
        	            				/home/runner/work/rook/rook/tests/integration/ceph_upgrade_test.go:104
        	Error:      	Not equal: 
        	            	expected: "rook/ceph:local-build"
        	            	actual  : "docker.io/rook/ceph:local-build"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-rook/ceph:local-build
        	            	+docker.io/rook/ceph:local-build
        	Test:       	TestCephUpgradeSuite/TestUpgradeHelm
```

**Comment by travisn** [comment]
@zalsader The CI is still failing, if you could take a look, thanks.

**Comment by zalsader** [comment]
I think my latest changes should fix it. Could you please rerun the failed tests?

**Comment by subhamkrai** [comment]
> I think my latest changes should fix it. Could you please rerun the failed tests?

re-started

**Comment by travisn** [comment]
@zalsader There is still an instance of this during the upgrade test:
```
    ceph_upgrade_test.go:383: 
        	Error Trace:	/home/runner/work/rook/rook/tests/integration/ceph_upgrade_test.go:383
        	            				/home/runner/work/rook/rook/tests/integration/ceph_upgrade_test.go:137
        	            				/home/runner/work/rook/rook/tests/integration/ceph_upgrade_test.go:104
        	Error:      	"rook/ceph:local-build" does not contain "docker.io/rook/ceph:local-build"
        	Test:       	TestCephUpgradeSuite/TestUpgradeHelm
```

**Comment by travisn** [comment]
@zalsader The CI looks good now! Can you just rebase it? Unfortunately there is a merge conflict now.

**Review by travisn — APPROVED** [review]

**Pull request merged**